ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerImpl.java

```diff
@@ -119,6 +119,12 @@
      */
     private final ReentrantLock inUseLock = new ReentrantLock();
 
+    /**
+     * The executing thread which is processing the timeout task This is only set to executing thread for TimerState.IN_TIMEOUT
+     * and TimerState.RETRY_TIMEOUT
+     */
+    private Thread executingThread = null;
```

Comment:
Should this be volatile? All other state in this class is final or volatile.

Reply:
Fixed to volatile.